db/repositories/profile_repository.py

```
@@ -0,0 +1,121 @@
+        Args:
+            profile: BlueskyProfile model to create or update
+            
+        Returns:
+            The created or updated BlueskyProfile object
+            
+        Raises:
+            ValueError: If profile.handle is empty
+            RuntimeError: If the profile cannot be created/updated due to a database error
+        """
+        if not profile.handle or not profile.handle.strip():
+            raise ValueError("profile.handle cannot be empty")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)